### PR TITLE
Clamp specification refactoring

### PR DESCRIPTION
**Pull request by ggalieroc**
Clamp specification refactored.

Details:

- Extended detailed description
- Modified the mathematical formula describing the op
- Modified range of values for attributes
- Added type of tensors
- Modified example to be more detailed

Ticket: 46103

### CODE REVIEW

**Review by vgavrilo — APPROVED** [review]

File: docs/ops/activation/Clamp_1.md [hunk 1]
```diff
@@ -6,48 +6,66 @@
...
+
 **Attributes**:
 
 * *min*
 
-  * **Description**: *min* is the lower bound of values in the output. Any value in the input that is smaller than the bound, is replaced with the *min* value. For example, *min* equal 10 means that any value in the input that is smaller than the bound, is replaced by 10.
-  * **Range of values**: non-negative positive floating point number
-  * **Type**: float
+  * **Description**: *min* is the lower bound of values in the output.
+  * **Range of values**: positive floating point number
```
Comment (ggalieroc) [inline docs/ops/activation/Clamp_1.md:30]: There is a list of [tests](https://github.com/openvinotoolkit/openvino/blob/master/ngraph/test/backend/fused_op.in.cpp#L483) which have negative min/max attributes, contradicting this statement:
- fused_clamp_double
- fused_clamp_float
- fused_clamp_int8
- fused_clamp_int16
- fused_clamp_int32
- fused_clamp_int64
- fused_clamp_uint8
- fused_clamp_uint16
- fused_clamp_uint32
- fused_clamp_uint64
- fused_clamp_float16
- fused_clamp_bfloat16

Other frameworks such as [pytorch](https://pytorch.org/docs/stable/generated/torch.clamp.html) and [onnx runtime](https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/test/providers/cpu/math/clip_test.cc) allow negative values for attributes.
@vgavrilo Do you have any comments here?

Comment (ggalieroc) [reply docs/ops/activation/Clamp_1.md]: After discussion we concluded that attributes may be negative.

**Review by vgavrilo — APPROVED** [review]

File: docs/ops/activation/Clamp_1.md [hunk 2]
```diff
@@ -6,48 +6,63 @@
...
   * **Default value**: None
   * **Required**: *yes*
 
 * *max*
 
-  * **Description**: *max* is the upper bound of values in the output. Any value in the input that is greater than the bound, is replaced with the *max* value. For example, *max* equals 50 means that any value in the input that is greater than the bound, is replaced by 50.
-  * **Range of values**: positive floating point number
-  * **Type**: float
+  * **Description**: *max* is the upper bound of values in the output.
+  * **Range of values**: arbitratry floating point number
```
Comment (lazarevevgeny) [inline docs/ops/activation/Clamp_1.md:34]: typo

File: docs/ops/activation/Clamp_1.md [hunk 3]
```diff
@@ -6,48 +6,63 @@
 
 **Short description**: *Clamp* operation represents clipping activation function.
 
+**Detailed description**:
+
+*Clamp* performs clipping operation over the input tensor element-wise. Element values of the output are within the range `[min, max]`.
+* Input values that are smaller than *min* are replaced with *min* value. For example, *min* equal 10 means that any value in the input that is smaller than the bound, is replaced by 10.
+* Input values that are greater than *max* are replaced with the *max* value. For example, *max* equals 50 means that any value in the input that is greater than the bound, is replaced by 50.
```
Comment (lazarevevgeny) [inline docs/ops/activation/Clamp_1.md:13]: I think that the explanation is clear enough and sentences "For example, *min* equal 10 means that any value in the input that is smaller than the bound, is replaced by 10." can be removed. The same for max

**Review by lazarevevgeny — APPROVED** [review]
Minor fixes are needed. Approved in advance.

File: docs/ops/activation/Clamp_1.md (hunk 3 above)
Comment (ggalieroc) [reply docs/ops/activation/Clamp_1.md]: I agree. Removed

File: docs/ops/activation/Clamp_1.md (hunk 2 above)
Comment (ggalieroc) [reply docs/ops/activation/Clamp_1.md]: Fixed

File: docs/ops/activation/Clamp_1.md [hunk 4]
```diff
@@ -6,48 +6,63 @@
 
 **Short description**: *Clamp* operation represents clipping activation function.
 
+**Detailed description**:
+
+*Clamp* performs clipping operation over the input tensor element-wise. Element values of the output are within the range `[min, max]`.
```
Comment (jdanieck) [inline docs/ops/activation/Clamp_1.md:11]: range `[min, max]`  -> **inclusive** range `[min, max]` ?

**Review by jdanieck — APPROVED** [review]

File: docs/ops/activation/Clamp_1.md (hunk 4 above)
Comment (ggalieroc) [reply docs/ops/activation/Clamp_1.md:11]: yes, inclusive range

**Pull request merged**